### DISCUSSION TITLE
Complex numbers in occa kernels

### DISCUSSION THREAD

**Discussion by E-Stroem** [libocca/occa#706] [Ideas]
Hello,
Is there a way to use C complex numbers and math inside an OCCA kernel?

E.g. the following C code in OCCA kernel:
double complex z1 = 1.0 + 2.0 * I
double complex z2 = 1.0 + 2.0 * I
double complex z3 = z1*z2;

and equivalent creal and cimag functions

Best regards,
E-Stroem

**Comment by dmed256** [comment]
Hi @E-Stroem,

Good question, I haven't thought about adding complex functions! 

One of the goals for the 2019 year is to add basic math methods that work on all backends. Like `max` would use `std::max` or just `max` for OpenCL/CUDA. Some support for that is already in OCCA kernels, but not everything. Adding complex methods like `creal` and `cimag` could be done in the same work.

I'm not too familiar with the standard complex methods, do you have a reference or list to some that might be handy?
